AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -208,74 +215,87 @@ - (void)migrateDatabase:(void *)__unused db fromVersion:(NSUInteger)__unused ver
+      long actualMaxSize = currentMaxPageCount * self.pageSize;
+      if (requestedMaxPageCount != currentMaxPageCount) {
+        MSLogError([MSAppCenter logTag], @"Could not change maximum database size to %ld bytes, current maximum size is %ld bytes.",
+                   sizeInBytes, actualMaxSize);
+        success = NO;
       } else {
-        MSLogInfo([MSAppCenter logTag], @"Changed maximum database size to %ld bytes (next multiple of 4KiB).", actualSize);
+        if (sizeInBytes == actualMaxSize) {
+          MSLogInfo([MSAppCenter logTag], @"Changed maximum database size to %ld bytes.", actualMaxSize);
+        } else {
+          MSLogInfo([MSAppCenter logTag], @"Changed maximum database size to %ld bytes (next multiple of 4KiB).", actualMaxSize);
+        }
+        self.maxSizeInBytes = actualMaxSize;
+        success = YES;
       }
-      self.maxPageCount = requestedMaxPageCount;
-      success = YES;
     }
   }
+  sqlite3_close(db);
   if (completionHandler) {
     completionHandler(success);
   }
-  sqlite3_close(db);
 }
 
-- (sqlite3 *)openDatabaseAtFileURL:(NSURL *)fileURL withMaxPageCount:(int)maxPageCount withResult:(int *)result {
++ (sqlite3 *)openDatabaseAtFileURL:(NSURL *)fileURL withResult:(int *)result {
   sqlite3 *db = NULL;
   *result = sqlite3_open_v2([[fileURL absoluteString] UTF8String], &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE | SQLITE_OPEN_URI, NULL);
   if (*result != SQLITE_OK) {
     MSLogError([MSAppCenter logTag], @"Failed to open database with result: %d.", *result);
     return NULL;
   }
-  NSString *statement = [NSString stringWithFormat:@"PRAGMA max_page_count = %i;", maxPageCount];
-  char *errorMessage;
-  *result = sqlite3_exec(db, [statement UTF8String], NULL, NULL, &errorMessage);
-  if (*result != SQLITE_OK) {
-    errorMessage = errorMessage ? errorMessage : "(nil)";
-    NSString *printableErrorMessage = [NSString stringWithCString:errorMessage encoding:NSUTF8StringEncoding];
-    MSLogError([MSAppCenter logTag], @"Failed to open database with specified maximum size constraint. Error message: %@",
-               printableErrorMessage);
-  }
   return db;
 }
 
-+ (int)numberOfPagesInBytes:(long)bytes {
-  return (int)(ceil((double)bytes / (double)kMSDefaultPageSizeInBytes));
++ (long)getPageSizeInOpenedDatabase:(void *)db {
+  NSArray<NSArray *> *rows = [MSDBStorage executeSelectionQuery:@"PRAGMA page_size;" inOpenedDatabase:db];
+  return [(NSNumber *)rows[0][0] longValue];
+}
+
++ (long)getPageCountInOpenedDatabase:(void *)db {
+  NSArray<NSArray *> *rows = [MSDBStorage executeSelectionQuery:@"PRAGMA page_count;" inOpenedDatabase:db];
+  return [(NSNumber *)rows[0][0] longValue];
+}
+
++ (long)getMaxPageCountInOpenedDatabase:(void *)db {
+  NSArray<NSArray *> *rows = [MSDBStorage executeSelectionQuery:@"PRAGMA max_page_count;" inOpenedDatabase:db];
+  return [(NSNumber *)rows[0][0] longValue];
+}
+
++ (int)setMaxPageCount:(long)maxPageCount inOpenedDatabase:(void *)db {
```

Comment:
Can't we use the `executeNonSelectionQuery:inOpenedDatabase:` here?

Reply:
Probably. That code was as it is, but I will consider this refactoring